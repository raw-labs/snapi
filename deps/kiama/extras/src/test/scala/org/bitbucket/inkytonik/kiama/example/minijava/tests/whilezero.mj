class WhileZero {
    public static void main () {
        System.out.println (new WhileZeroClass ().run ());
    }
}

class WhileZeroClass {

    public int run () {
        int v;
        v = 0;
        while (false) {
            v = v + 1;
        }
        return v;
    }

}
