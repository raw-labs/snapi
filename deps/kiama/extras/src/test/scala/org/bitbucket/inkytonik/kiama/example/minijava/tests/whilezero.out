.source src/test/scala/org/bitbucket/inkytonik/kiama/example/minijava/tests/whilezero.mj
.class public WhileZero
.super java/lang/Object

.method <init>()V
.limit stack 1
.limit locals 1
    aload_0
    invokespecial java/lang/Object/<init>()V
    return
.end method

.method public static main([Ljava/lang/String;)V
.limit stack 3
.limit locals 2
    getstatic java/lang/System/out Ljava/io/PrintStream;
    new WhileZeroClass
    dup
    invokespecial WhileZeroClass/<init>()V
    invokevirtual WhileZeroClass/run()I
    invokevirtual java/io/PrintStream/println(I)V
    return
.end method

.source src/test/scala/org/bitbucket/inkytonik/kiama/example/minijava/tests/whilezero.mj
.class public WhileZeroClass
.super java/lang/Object

.method <init>()V
.limit stack 1
.limit locals 1
    aload_0
    invokespecial java/lang/Object/<init>()V
    return
.end method

.method public run()I
.limit stack 2
.limit locals 2
    iconst_0
    istore 1
L1:
    iconst_0
    ifeq L2
    iload 1
    iconst_1
    iadd
    istore 1
    goto L1
L2:
    iload 1
    ireturn
.end method

